src/udev/udev-builtin-path_id.c

```diff
@@ -347,24 +347,24 @@ static sd_device *handle_scsi_default(sd_device *parent, char **path) {
                 return NULL;
 
         FOREACH_DIRENT_ALL(dent, dir, break) {
-                char *rest;
-                int i;
+                unsigned i;
+                int r;
 
                 if (dent->d_name[0] == '.')
                         continue;
                 if (!IN_SET(dent->d_type, DT_DIR, DT_LNK))
                         continue;
                 if (!startswith(dent->d_name, "host"))
                         continue;
-                i = strtoul(&dent->d_name[4], &rest, 10);
-                if (rest[0] != '\0')
+                r = safe_atou_full(&dent->d_name[4], 10, &i);
+                if (r < 0)
```

Comment:
`r` is not used, so `if (safe_atou_full(...) < 0)` is ok.

Reply:
Fixed.